src/commonMain/kotlin/exchange.dydx.abacus/state/v2/supervisor/account/PushNotificationRegistrationHandler.kt

```diff
@@ -0,0 +1,97 @@
+package exchange.dydx.abacus.state.v2.supervisor.account
+
+import exchange.dydx.abacus.protocols.TransactionType
+import exchange.dydx.abacus.state.v2.supervisor.NetworkHelper
+import exchange.dydx.abacus.utils.Logger
+import exchange.dydx.abacus.utils.iMapOf
+import exchange.dydx.abacus.utils.toJsonPrettyPrint
+
+internal interface PushNotificationRegistrationHandlerProtocol {
+    fun sendPushNotificationToken(token: String, languageCode: String, isKepler: Boolean)
+}
+
+internal class PushNotificationRegistrationHandler(
+    private val helper: NetworkHelper,
+    private val accountAddress: String,
+) : PushNotificationRegistrationHandlerProtocol {
+
+    override fun sendPushNotificationToken(token: String, languageCode: String, isKepler: Boolean) {
+        getSignedPayload(token, languageCode, isKepler) { payload ->
+            if (payload != null) {
+                sendPushNotificationToken(payload)
+            } else {
+                Logger.e { "Push notification token payload signing failed" }
+            }
+        }
+    }
+
+    private fun sendPushNotificationToken(
+        payload: Map<String, Any>,
+    ) {
+        val indexerUrl = helper.configs.indexerConfig?.api ?: return
+        val registrationUrl = "$indexerUrl/v4/addresses/$accountAddress/registerToken"
+        val header =
+            iMapOf(
+                "Content-Type" to "application/json",
+            )
+        helper.post(
+            url = registrationUrl,
+            headers = header,
+            body = payload.toJsonPrettyPrint(),
+        ) { _, response, httpCode, _ ->
+            if (helper.success(httpCode) && response != null) {
+                Logger.d { "Push notification token registered successfully" }
+            } else {
+                Logger.e { "Push notification token registration failed" }
+            }
+        }
+    }
+
+    private fun getSignedPayload(
```

Comment:
this could probably be a suspending function instead - in general i think we should move to suspending functions over callbacks.

Reply:
This goes though the JS bridging function, and it took a lot of effort to get it right the first time.  I don't think we should spending time re-do this now.